crates/tx-parser/src/filter_types.rs

```diff
@@ -0,0 +1,61 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use strata_primitives::{
+    buf::Buf32,
+    l1::BitcoinAddress,
+    params::{DepositTxParams, RollupParams},
+    sorted_vec::SortedVec,
+};
+
+use crate::utils::{generate_taproot_address, get_operator_wallet_pks};
+
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize)]
+pub struct TxFilterConfig {
+    /// For checkpoint update inscriptions.
+    pub rollup_name: String,
+
+    /// For addresses that we expect spends to.
+    pub expected_addrs: SortedVec<BitcoinAddress>,
+
+    /// For blobs we expect to be written.
+    pub expected_blobs: SortedVec<Buf32>,
+
+    /// For deposits that might be spent from.
+    pub expected_outpoints: SortedVec<Outpoint>,
+
+    /// Deposit config that defines the structure we expect in the utxo
+    pub deposit_config: DepositTxParams,
+}
+
+impl TxFilterConfig {
+    // TODO: this will need chainstate too in the future
+    pub fn derive_from(rollup_params: &RollupParams) -> anyhow::Result<Self> {
+        let operator_wallet_pks = get_operator_wallet_pks(rollup_params);
+        let address = generate_taproot_address(&operator_wallet_pks, rollup_params.network)?;
+
+        let rollup_name = rollup_params.rollup_name.clone();
+        let expected_blobs = SortedVec::new(); // TODO: this should come from chainstate
+        let expected_addrs = SortedVec::from(vec![address.clone()]);
+        let expected_outpoints = SortedVec::new();
+
+        let deposit_config = DepositTxParams {
+            magic_bytes: rollup_name.clone().into_bytes(),
+            address_length: rollup_params.address_length,
+            deposit_amount: rollup_params.deposit_amount,
+            address,
+        };
+        Ok(Self {
+            rollup_name,
+            expected_blobs,
+            expected_addrs,
+            expected_outpoints,
+            deposit_config,
+        })
+    }
+}
+
+/// Outpoint of a bitcoin tx
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, BorshSerialize, BorshDeserialize)]
+pub struct Outpoint {
+    pub txid: Buf32,
+    pub vout: u32,
+}
```

Comment:
This should be moved to `crates/primitives/src/l1.rs`.
See example of `BitcoinAddress`: https://github.com/alpenlabs/strata/blob/77cae83ebf5319b17ae10fa8ce5df5876f2dbec0/crates/primitives/src/l1.rs#L236-L247

Reply:
I am not sure about this. The Outpoint is supposed to be more "local" than the BitcoinAddress. That's why it is here.

Reply:
I think that this invites code duplication.
It is way easier to have it in `strata/crates/primitives/src/l1.rs` so that we can use it in the future if necessary.
Especially if we want to `Borsh` stuff and with the `Buf32`.